test/EngineTest.cpp

```diff
@@ -43,94 +57,259 @@ IdTable makeIdTableFromVector(std::vector<std::array<size_t, TABLE_WIDTH>> table
+ *  is sorted by the join column.
+ * @param jc The join column of the table.
+ * @param T Ignore it. It's only here for being able to make better messages,
+ *  if a IdTable fails the comparison.
+*/
+template<size_t TABLE_WIDTH>
+void compareIdTableWithSolution(IdTable table, 
+    std::vector<std::array<size_t, TABLE_WIDTH>> sampleSolution,
+    const bool testForSorted = false,
+    const size_t jc = 0,
+    ad_utility::source_location T = ad_utility::source_location::current()
+    ) {
+  // For generating more informative messages, when failing the comparison.
+  std::stringstream traceMessage;
+  traceMessage << "compareIdTableWithSolution comparing IdTable\n";
+  for (size_t row = 0; row < table.size(); row++) {
+    for (size_t column = 0; column < table.cols(); column++) {
+      traceMessage << table(row, column) << " ";
+    }
+    traceMessage << "\n";
+  }
+  traceMessage << "with IdTable described by vector\n";
+  for (auto const& row: sampleSolution) {
+    for (auto const& entry: row) {
+      traceMessage << entry << " ";
+    }
+    traceMessage << "\n";
+  }
+  auto trace = generateLocationTrace(T, traceMessage.str());
+  
+  // Do the IdTable and sampleSolution have the same dimensions?
+  ASSERT_EQ(table.size(), sampleSolution.size());
+  ASSERT_EQ(table.cols(), TABLE_WIDTH);
+
+  if (testForSorted) {
+    // Is the table sorted by join column?
+    auto oldEntry = table(0, jc);
+    for (size_t i = 1; i < table.size(); i++) {
+      ASSERT_TRUE(oldEntry <= table(i, jc));
+      oldEntry = table(i, jc);
+    }
   }
-  a.push_back({I(400000), I(200000)});
-  b.push_back({I(400000), I(200000)});
 
-  CALL_FIXED_SIZE_3(lwidth, rwidth, reswidth, J.join, a, 0, b, 0, &res);
-  ASSERT_EQ(6u, res.size());
+  // Sort both the table and the sampleSolution, so that both have a definite
+  // form for comparison.
+  auto sortFunction = [](
+      const auto& element1,
+      const auto& element2)
+    {
+      size_t i = 0;
+      while (i < (TABLE_WIDTH - 1) && element1[i] == element2[i]) {
+        i++;
+      }
+      return element1[i] < element2[i];
+    };
+  std::sort(table.begin(), table.end(), sortFunction);
+  std::sort(sampleSolution.begin(), sampleSolution.end(), sortFunction);
+
+  for (size_t row = 0; row < sampleSolution.size(); row++) {
+    for (size_t column = 0; column < TABLE_WIDTH; column++) {
+      ASSERT_EQ(I(sampleSolution[row][column]), table(row, column));
+    }
+  }
+}
 
-  a.clear();
-  b.clear();
-  res.clear();
+/*
+ * @brief Converts the given vectors to IdTables, joins them together with the
+ * given join function and returns the result.
+ *
+ * @tparam TABLE_A_WIDTH is the width of the first table.
+ * @tparam TABLE_B_WIDTH is the width of the second table.
+ * @tparam JOIN_FUNCTION is used to allow the transfer of any type of
+ *  lambda function, that could contain a join function. You never have to
+ *  actually specify this parameter , just let inference do its job.
```

Comment:
I think you don't need to document those, as they are absolutely clear when you look at the arguments' types and documentation.

Reply:
The referred to function is no longer a generic function and no longer possesses this part of the documentation.